ratelimitbackend/backends.py

```diff
@@ -19,9 +19,13 @@ class RateLimitMixin(object):
     minutes = 5
     requests = 30
     username_key = 'username'
+    no_username = False
 
     def authenticate(self, request=None, **kwargs):
-        username = kwargs[self.username_key]
+        username = kwargs.get(self.username_key)
+        if username is None and not self.no_username:
+            raise KeyError
```

Comment:
Would this achieve the same thing?

```python
username = None
try:
    username = kwargs[self.username_key]
except KeyError:
    if not self.no_username:
        raise
```

with the benefit of raising KeyError with the relevant key.